src/s3.rs

```diff
@@ -0,0 +1,211 @@
+use aws_sdk_lambda::types::ByteStream;
+use aws_sdk_s3::{
+    model::{Delete, ObjectIdentifier, Tag, Tagging},
+    RetryConfig,
+};
+use aws_types::SdkConfig;
+
+use crate::errors::AwsError;
+
+pub struct S3Client {
+    client: aws_sdk_s3::Client,
+}
+
+#[derive(Debug, Clone)]
+pub struct OutputS3ObjectMetadata {
+    pub key: String,
+    pub size: i64,
+    pub last_modified: i64,
+}
+
+impl S3Client {
+    pub fn new(config: &SdkConfig) -> S3Client {
```

Comment:
```suggestion
    /// Returns a S3 client built from `config`.
    ///
    /// In the retry config for requests, the maximum number of attempts is set to 5.
    pub fn new(config: &SdkConfig) -> S3Client {
```

More rust doc would be nice, but I think this part is very important to have.